barretenberg/cpp/pil/vm2/execution.pil

```diff
@@ -746,9 +746,9 @@ sel_exec_dispatch_returndata_copy {
 // Note that we enforce the output tag through this dispatching lookup by having ia_tag matching both mem_tag_reg[0] and rop[1].
 #[DISPATCH_TO_SET]
 sel_exec_dispatch_set {
-    rop[2], rop[1], subtrace_operation_id, register[0], mem_tag_reg[0], sel_opcode_error
-} in alu.sel_op_truncate {
-    alu.ia, alu.ia_tag, alu.op_id, alu.ic, alu.ia_tag, /*sel_opcode_error=*/ precomputed.zero
+    rop[2], rop[1], register[0], mem_tag_reg[0], subtrace_operation_id, sel_opcode_error
+} in alu.sel {
+    alu.ia, alu.ia_tag, alu.ic, alu.ia_tag, alu.op_id, /*sel_opcode_error=*/ precomputed.zero
```

Comment:
Could we not keep the gating by `alu.sel_op_truncate` and look up something like `precomputed.one` against `alu.sel`? I'm not sure if it would be lower overhead (or if we care here), but may be clearer as the use of `alu.sel` here implies to the reader it should apply to all on alu rows. Maybe just a comment to explain would be fine!